velox/type/Conversions.h

```diff
@@ -383,6 +383,38 @@ struct Converter<TypeKind::TIMESTAMP> {
   static T cast(const std::string& v, bool& nullOutput) {
     return fromTimestampString(v);
   }
+
+  static T cast(const Date& d, bool& nullOutput) {
+    static const int32_t kMillisPerDay{86'400'000};
+    return Timestamp::fromMillis(d.days() * kMillisPerDay);
```

Comment:
maybe we could have a `Timestamp::fromDays()` method?

Reply:
I will do a bit more cleanup in my next PR. Will address it there.